src/plugins/interaction/ROSClockServer/ROSClockServer.cpp

```diff
@@ -73,16 +73,40 @@ bool ROSClockServer::init(std::map<std::string, std::string> &mission_params,
         ros::init(argc, NULL, "scrimmage", ros::init_options::NoSigintHandler);
     }
 
-    nh_ = std::make_shared<ros::NodeHandle>();
-    clock_pub_ = nh_->advertise<rosgraph_msgs::Clock>("/clock", 1);
+    // rosmaster check thread
+    check_rosmaster_thread_ = std::thread(&ROSClockServer::check_rosmaster, this);
+    running_ = true;
```

Comment:
`std::thread` begins execution immediately upon construction, but running_ is set to true after construction.  This could result in a race condition where `while(running_)` evaluates to false and nothing happens in the thread before it returns.  Please consider moving `running_ = true;` to the line before the thread initialization.